ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/KsqlKeyExtractor.java

```diff
@@ -19,26 +19,60 @@
 
 import io.confluent.ksql.GenericKey;
 import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.transform.ExpressionEvaluator;
+import io.confluent.ksql.logging.processing.ProcessingLogger;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 public final class KsqlKeyExtractor<KRightT> implements Function<GenericRow, KRightT> {
 
   private final int leftJoinColumnIndex;
 
+  private final Optional<ExpressionEvaluator> expressionEvaluator;
+  private final Optional<ProcessingLogger> processingLogger;
+  private final Optional<Supplier<String>> errorMessage;
+
   KsqlKeyExtractor(final int leftJoinColumnIndex) {
     checkArgument(
         leftJoinColumnIndex >= 0,
         "leftJoinColumnIndex negative: " + leftJoinColumnIndex
     );
 
     this.leftJoinColumnIndex = leftJoinColumnIndex;
+    expressionEvaluator = Optional.empty();
```

Comment:
As above, it'd be nice if we could simplify this code by always using an expression evaluator.